src/components/Nav/useNav.ts

```diff
@@ -26,15 +27,11 @@ import { trackCustomEvent } from "@/lib/utils/matomo"
 
 import type { NavSections } from "./types"
 
-import useColorModeValue from "@/hooks/useColorModeValue"
-
 export const useNav = () => {
   const { t } = useTranslation("common")
-  const { theme, setTheme } = useTheme()
+  const { setTheme, resolvedTheme, systemTheme } = useTheme()
```

Comment:
Removed usage of `theme` which was not always responding appropriately to system color preference changes.

In it's place, `resolvedTheme` is a more reliable "light" or "dark" string that the theme will resolve to. Also, `systemTheme` taps into the color mode preference of the system, and updates on change.